### PR TITLE
Convert to Jest

### PR DESCRIPTION
**Pull request by orta**
Switches our test runner from Mocha to Jest, as they both have the same things under the hood, a lot of the changes end up being cosmetic main `before` -> `beforeAll`.

### CODE REVIEW

**Comment by alloy** [comment]
Nice work! 👏 

So this still uses `expect.js` instead of Jest’s own `expect` matchers, right? Does Jest have matchers that work with Promises, similar to the ones we now define in the test helper?

**Comment by orta** [comment]
Yes, this was the MVP of moving over. 

WRT matchers, I'm not 100%, they imply that if you return a promise then it will wait for it all to finish up before wrapping the test:

```
describe('my beverage list', () => {
  it('has lemon in it', () => {
    return fetchBeverageList().then((list) => {
      expect(list).toContain('lemon');
    });
  });
});
```

But I don't think that's what you're after. It sits on top of Jasmine if I recall, so I would expect something like [promise-matchers](https://www.npmjs.com/package/promise-matchers) to work

File: package.json [hunk 1]
```diff
@@ -60,15 +61,22 @@
...
+    "babel-jest": "^16.0.0",
     "babel-plugin-rewire": "1.0.0-beta-2",
     "deep-equal": "^1.0.1",
     "eslint": "^1.10.3",
     "eslint-config-airbnb": "^3.1.0",
     "eslint-plugin-react": "^3.14.0",
     "expect.js": "^0.3.1",
     "foreman": "^1.4.1",
+    "jest": "^16.0.2",
     "mocha": "^2.3.3",
```
Comment (alloy) [inline package.json]: This should be removed.

Comment (orta) [inline package.json]: oh, weird, I thought `npm uninstall` would do that, guess I had to save or something

**Comment by orta** [comment]
👍  done'd

**Comment by alloy** [comment]
Right, gotcha, that should work then. Agreed on MVP, let’s make the switch 👍

**Pull request merged**